lib/util/git/index.ts

```diff
@@ -215,6 +217,23 @@ export async function setBranchPrefix(branchPrefix: string): Promise<void> {
   }
 }
 
+function setIgnoredAuthors(ignoredAuthors?: string[]): void {
+  config.ignoredAuthors = ignoredAuthors ?? [];
+}
+
+export interface UserRepoConfig {
+  branchPrefix: string;
+  gitIgnoredAuthors: string[];
+}
+
+export async function setUserRepoConfig({
+  branchPrefix,
+  gitIgnoredAuthors,
+}: Partial<UserRepoConfig>): Promise<void> {
```

Comment:
I would like to make it non-`Partial`, but those fields are optional

Reply:
This seems to be a recurring challenge with config which is user-derived. Although internally we should assume such fields are mandatory, they originate as optional in user config prior to us massaging to defaults if empty. But we also have the complication where users *can* configure `null` and override a non-null default. I wonder if we could remove `null` as a valid value in options, e.g. like we did some times by changing null/false/true to be auto/disabled/enabled.

Reply:
```suggestion
}: RenovateConfig): Promise<void> {
```

Why not simply reuse `RenovateConfig` interface?